src/token/governance/ERC20Base.sol

```diff
@@ -61,6 +61,9 @@ contract ERC20Base is
     {
         require(admin != address(0), "Admin address cannot be zero");
 
+        bool tokenNameEmpty = keccak256(abi.encode(tokenName)) == keccak256(abi.encode(""));
+        bool tokenSymbolEmpty = keccak256(abi.encode(tokenSymbol)) == keccak256(abi.encode(""));
+        require(!(tokenNameEmpty && tokenSymbolEmpty), "ERC20Base: Token name cannot be empty");
```

Comment:
This check both `name` and `symbol`, but the reason string only mentions `name`. It would be better to have two requires, each with their own reason string.

Entirely out of curiosity, why `keccak256(abi.encode(tokenName))` and not `keccak256(bytes(tokenName))`? Is it more efficient since it's `bytes32` and not `bytes`?

Reply:
@mrmemes-eth 
Part 1, was just a miss on my part. Fixing
Part 2, I actually was going to use `encodedPacked` because that was in the Solidity docs but then read that it was "nonstandard" so thought I'd use encode instead. I tried to find answers to which was better as I saw both in various  examples and wasn't able to get a clear answer so I went with the one closest to the docs. So the upshot is .. I am actually not sure which is better. I did spend a fair amount of time trying to understand it and googling but I am still a bit unsure. Would love your thoughts and opinion.

 Here is what it says: 
"Solidity does not have string manipulation functions, but there are third-party string libraries. You can also compare two strings by their keccak256-hash using keccak256(abi.encodePacked(s1)) == keccak256(abi.encodePacked(s2)) and concatenate two strings using string.concat(s1, s2).

You should use bytes over bytes1[] because it is cheaper, since using bytes1[] in memory adds 31 padding bytes between the elements. Note that in storage, the padding is absent due to tight packing, see [bytes and string](https://docs.soliditylang.org/en/develop/internals/layout_in_storage.html#bytes-and-string). As a general rule, use bytes for arbitrary-length raw byte data and string for arbitrary-length string (UTF-8) data. If you can limit the length to a certain number of bytes, always use one of the value types bytes1 to bytes32 because they are much cheaper."